packages/gatsby/cache-dir/production-app.js

```diff
@@ -2,6 +2,7 @@ import { apiRunner, apiRunnerAsync } from "./api-runner-browser"
 import React, { createElement } from "react"
 import ReactDOM from "react-dom"
 import { Router, navigate } from "@reach/router"
+import { match } from "@reach/router/lib/utils"
```

Comment:
maybe not worth using the private require here? It might be enough to just bail when matchPath exists at all?

Reply:
maybe a more complete check would be to replace the part before the `*` in the matchPath with the canonical page path, e.g. so `matchPath: '/foo/bar/*'` with a client pathname of `/Foo/Bar/baz` is redirected to `/foo/bar/baz` but idk know to do that :P

Reply:
We use some private(ish) utils already - https://github.com/gatsbyjs/gatsby/blob/4c676f53df4bd9453dc82f898307d9ef26d1970e/packages/gatsby-react-router-scroll/src/ScrollBehaviorContext.js#L4 so I think that's fine